devolutions-gateway/src/streaming/mod.rs

```diff
@@ -0,0 +1,58 @@
+use std::sync::{atomic::AtomicBool, Arc};
+
+use anyhow::Context;
+use axum::{body::Body, response::Response};
+use streamer::streamer::{reopenable_file::ReOpenableFile, webm_stream};
+use uuid::Uuid;
+
+use crate::{recording::OnGoingRecordingState, token::RecordingFileType, ws::websocket_compat};
+
+pub(crate) async fn stream_file(
+    path: camino::Utf8PathBuf,
+    ws: axum::extract::WebSocketUpgrade,
+    shutdown_signal: devolutions_gateway_task::ShutdownSignal,
+    recordings: crate::recording::RecordingMessageSender,
+    recording_id: Uuid,
+) -> anyhow::Result<Response<Body>> {
+    // 1.identify the file type
+    if !(path.extension() == Some(RecordingFileType::WebM.extension())) {
+        return Err(anyhow::anyhow!("Invalid file type"));
+    }
+    // 2.if the file is actively being recorded, then proceed
+    let Ok(Some(OnGoingRecordingState::Connected)) = recordings.get_state(recording_id).await else {
+        return Err(anyhow::anyhow!("File is not being recorded"));
+    };
+
+    let streaming_file = ReOpenableFile::open(&path).with_context(|| format!("Failed to open file: {:?}", path))?;
```

Comment:
style: https://github.com/Devolutions/IronRDP/blob/87014d4afb3229813f4a61ab30fc7afd778e3fb4/STYLE.md#formatting-of-error-messages
```suggestion
    let streaming_file = ReOpenableFile::open(&path).with_context(|| format!("failed to open file: {path:?}"))?;
```